correct spells

tarotene opened the pull request.

jhqian committed:
dts: arm: nxp: update rt1170 dts

- update soc dtsi

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
soc: arm: nxp_imx: add rt117x support

1. Added RT10xx and RT11xx configs
2. Added a new soc file for rt117x. There are clock differences
   between the RT10xx and RT11xx series, hence the soc files
   have been separated.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
drivers: clock_control: add ccm driver for rt117x

- add ccm driver for rt117x

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
board: arm: add board support files for mimxrt1170_evk

Add board support files for mimxrt1170_evk

- Add pinmux, dts and doc
- Tested samples: hello_world, philosophers, synchronization,
  basic/blinky, basic/button.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

lucien-nxp committed:
west: update the head of hal_nxp

- to get the update hal_nxp from PR #76

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

Commit:
esp32: cache: test: add Flash/SPIRAM cache test

Add SPI Flash/SPI RAM coexistence test.

Signed-off-by: Glauber Maroto Ferreira <glauber.ferreira@espressif.com>

finikorg committed:
edac: Introduce enum type for error type

Use edac_error_type enum type instead of defines for EDAC error types.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Refactor EDAC API

Refactor EDAC API making it more clear, removing unneeded typedefs and
using check for optional and assert for mandatory APIs.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: shell: Use new EDAC API in shell

Use new EDAC API in shell.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: tests: Use new EDAC API in tests

Use new EDAC API in tests.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Include error injection API to documentation

Remove #ifdefs making code cleaner and including error injection API
functions to documentation.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: docs: Use @retval instead of @return when applicable

Make documentation more readable by using @retval instead of @return
when applicable.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Use -ENOSYS for not implemented mandatory API functions

Use standard Zephyr way for not implemented mandatory API functions.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Use TODO instead of workaround in comment

Using TODO is more appropriate since workaround implies working around
a bug.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: shell: Replace ret with err

Replace ret with err for simple int type errors.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: shell: Use boolean expressions in if conditions

Use only boolean in if conditions as new coding style implies.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: ibecc: Use boolean expressions in if conditions

Use only boolean in if conditions as new coding style implies.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: ibecc: Simplify NMI handling

Refactor NMI handling making it clearer and fix return code for other
NMI source.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: ibecc: Simplify PCI probing

Use only one pcie_conf_read() instead of several reads and make code
more readable.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Use boolean expressions in if conditions

Use only boolean in if conditions as new coding style implies for edac
API header file.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: Use -ENOSYS also for optional interfaces

Use -ENOSYS for all not implemented interfaces.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

LingaoM committed:
Bluetooth: Mesh: Check the CID field before opcode compare

Bluetooth Mesh Vendor model hava company id field.

Accordin MeshPRFV1.0.1 3.7.3.1 Operation codes.

The 3-octet opcodes are used for manufacturer-specific opcodes.
The company identifiers are 16-bit values defined by the
Bluetooth SIG and are coded into the second and third octets of
the 3-octet opcodes.

Therefore, we can speed up the search process by checking whether
CID fields match, rather than comparing opcodes one by one.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

finikorg committed:
edac: Add dumy EDAC driver for API coverage

Add dummy EDAC driver for -ENOSYS test coverage.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: ibecc: Strip debug information

Remove debug information increasing code coverage.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

avisconti committed:
dts/bindings: lis2dw12: create a common st,lis2dw12-common.yaml

Create a common binding file that will be included by all bindings
handled by lis2dw12 driver.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: move int-pin in DTS binding

Take the int-pin information (i.e. what pin between INT1
and INT2 the drdy is attached to) directly from DT.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: use gpio_dt_spec oriented macros

Use gpio_dt_spec structure (and related APIs) in config for configuring
the gpio used for drdy and pulse interrupts.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Move trigger pulse Kconfig property into DT

Move lis2dw12 trigger pulse configurations from Kconfigs to Device Tree.
Moreover the dts properties have been renamed as 'tap', which sounds a
better name to immediately catch the feature behind it. Since tap
threshold cannot be zero, this value (which is the default in dts
binding) is used to enable/disable the device feature per each axis.
The event can be generated on INT1 only.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

de-nordic committed:
dfu: flash_img: Removing conditional compilation from header

Applying Coding Guideline rule A.1: Conditional Compilation.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

avisconti committed:
drivers/sensor: lis2dw12: Move power Kconfig property into dts

Move lis2dw12 power-mode option from Kconfigs to Device Tree.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Move range Kconfig property into dts

Converts lis2dw12 range options (2g, 4g, 8g, 16g) from Kconfigs
to Device Tree.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Remove odr values from Kconfig

Remove all odr values from Kconfig and always init it
at 12.5Hz. It is responsibility of application to set
the rate to a different value using SENSOR_ATTR_SAMPLING_FREQUENCY.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Add multi-instance support

This commit aligns lis2dw12 sensor driver to latest multi
instance sensor driver model.

In particular it makes use of the stmemsc common routines
and move ctx handler inside struct config, so that the
bus_init routines can be totally avoided.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Fix the logic to set TAP interrupt

The lis2dw12 sensor can generate the TAP interrupt only on INT1,
while DRDY can be generated on both. The int-pin DT property
specifiy on which pin the DRDY (and not the TAP) can be generated.

This commit fix the way the trigger is set: first the driver checks
the trigger type (DRDY or TAP), then it uses the int-pin information
only in DRDY case but allows setting TAP regardless of int-pin (it
always routes it on INT1).

The previous code was first checking int-pin: if it was INT2 then
the driver refused setting TAP triggers.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: lis2dw12: Set tap triggers only if detection is on

Set single/double tap triggers only if tap detection is enabled.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

fabiobaltieri committed:
dts: stm32f4: add fmc device node for f4 series devices

Add a device node for the FMC controller found in stm32f427, f437 and
f446 devices, works fine with the current memc_stm32_sdram driver.

Signed-off-by: Fabio Baltieri <fabio.baltieri@gmail.com>

fabiobaltieri committed:
boards: stm32f429i_disc1: add sdram support

Add support for the 8MB SDRAM found in STM32F429-DISC1 boards.

Signed-off-by: Fabio Baltieri <fabio.baltieri@gmail.com>

ABOSTM committed:
boards: arm: stm32h747i_disco: add Arduino SPI support

Add Arduino SPI support for stm32h747i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f769i_disco: add Arduino SPI support

Add Arduino SPI support for stm32f769i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f723e_disco: add Arduino SPI support

Add Arduino SPI support for stm32f723e_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f469i_disco: add Arduino SPI support

Add Arduino SPI support for stm32f469i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l476rg: add Arduino SPI support

Add Arduino SPI support for nucleo_l476rg.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l433rc_p: add Arduino SPI support

Add Arduino SPI support for nucleo_l433rc_p.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_g431rb: fix Arduino SPI support

Fix Arduino SPI support for nucleo_g431rb.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_g474re: fix Arduino SPI support

Fix Arduino SPI support for nucleo_g474re.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ppryga-nordic committed:
samples: Bluetooth: df_rx: Add config for support the nRF52820

Add required configuration and DTS overlay for support the nRF52820 SOC.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

finikorg committed:
edac: ibecc: Reduce CCM number using fallthrough

Reduce CCM number reported by Coverity by using __fallthrough.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

Thalley committed:
Bluetooth: ISO: Only remove CIG as central

The hci_le_remove_cig command shall only be sent as the
master/central. Implemented this by early termination in
bt_iso_cleanup as the slave/peripheral.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

mmahadevan108 committed:
drivers: hwinfo: Update i.MX RT Hwinfo driver

Add support for RT1170. The fuse register that holds
the unique device ID is different as compared to the
RT10XX series

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

fabiobaltieri committed:
drivers: stm32-fmc: enable MEMC_STM32 default based on dt

Default the option MEMC_STM32 to "y" when the device node is defined and
enabled, so that the driver is selected automatically when the board
supports it and MEMC is enabled.

Remove the default conditional on serise as it's redundant with the one
in soc/arm/st_stm32/common/Kconfig.defconfig.series.

Signed-off-by: Fabio Baltieri <fabio.baltieri@gmail.com>

ABOSTM committed:
boards: arm: b_l4s5i_iot01a: full Arduino SPI support

Full Arduino SPI support for b_l4s5i_iot01a

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: disco_l475_iot1: full Arduino SPI support

Full Arduino SPI support for disco_l475_iot1

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f030r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f030r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f070rb: full Arduino SPI support

Full Arduino SPI support for nucleo_f070rb

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f103rb: full Arduino SPI support

Full Arduino SPI support for nucleo_f103rb

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f207zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f207zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f302r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f302r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f334r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f334r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f401re: full Arduino SPI support

Full Arduino SPI support for nucleo_f401re

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32l496g_disco: full Arduino SPI support

Full Arduino SPI support for stm32l496g_disco

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f412zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f412zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f413zh: full Arduino SPI support

Full Arduino SPI support for nucleo_f413zh

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f429zi: full Arduino SPI support

Full Arduino SPI support for nucleo_f429zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f746zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f746zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f756zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f756zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f767zi: full Arduino SPI support

Full Arduino SPI support for nucleo_f767zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_h743zi: full Arduino SPI support

Full Arduino SPI support for nucleo_h743zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_h753zi: full Arduino SPI support

Full Arduino SPI support for nucleo_h753zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l412rb_p: full Arduino SPI support

Full Arduino SPI support for nucleo_l412rb_p

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l452re: full Arduino SPI support

Full Arduino SPI support for nucleo_l452re

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l496zg: full Arduino SPI support

Full Arduino SPI support for nucleo_l496zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l4r5zi: full Arduino SPI support

Full Arduino SPI support for nucleo_l4r5zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f746g_disco: full Arduino SPI support

Full Arduino SPI support for stm32f746g_disco

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: shield: x_nucleo_idb05a1: add nucleo_l476rg spi overlay

Standard Arduino CLK signal is on D13,
nucleo_l476rg board has been modified accordingly.
Need an overlay to use default x_nucleo_idb05a1 spin D3 (PB3)

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32h747i_disco: rework uart split between CM4 and CM7

Move uart/usart pinctrl to common file.
Set status in dedicated m7 or m4 file.
But only one core can control a peripheral instance.
Arbitrary default choice:
usart1 --> m7
uart8 --> m4

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ycsin committed:
dts/arm: stm32g0: Add lptim1

Add lptim1 node for the G0 series.

Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

mmahadevan108 committed:
dt-bindings: clock: Update for i.MX CCM Rev 2

- update clock macro for CCM
- Add a dts bindings file for CCM Rev 2

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

finikorg committed:
edac: shell: Check error for log clear functions

Check errors returned by edac_ecc_error_log_clear() and
edac_parity_error_log_clear().

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: shell: Print exact error with shell error message

Include to shell_error() also exact error.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
edac: shell: Use helper functions for shell devmem command

Using helper functions decreases function complexity numbers reported
by static code analyzers.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

Commit:
drivers: ps2: Fix syscalls handlers

This driver was using a very old macro to declare a syscall. Fix
syscall handlers.

Fixes #23745

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
drivers: ps2: Fix wrong cast

ps2_read() syscall was casting the value to uint32_t *

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
drivers: ps2: Build syscall handlers

syscall handler were not being built at all !!

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nordic-krch committed:
drivers: serial: nrfx_uarte: Fix race condition in async isr

RXTO event is generated always after ENDRX and driver relies
on assumption that ENDRX event is handled before RXTO. However,
when interrupt is preempted after ENDRX check returned false
and RXTO event is already set handling order would be swapped.

Added addtional check to handle RXTO event only if ENDRX is not
set. If ENDRX is set, it means that it is not yet handled. RXTO
event is not cleared and interrupt will be triggered again and
ENDRX event will be handled first.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

Commit:
drivers: display: ili9xxx: add support for ILI9341 controller

Add support for the ILI9341 display controller.

Signed-off-by: Krivorot Oleg <krivorot.oleg@gmail.com>

rakons committed:
sys: printk: Fix LOG2 printk support in cpp code

This commit fixes the issue with a compilation of the sample that
uses printk function from cpp code when LOG2 is used and printk
is handled by logging subsystem.

Signed-off-by: Radoslaw Koppel <radoslaw.koppel@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Minor change to use IS_ENABLED

Minor changes to use IS_ENABLED and updates to comments in
code.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix Data Length Update node Rx reservations

Reserve a minimum node rx of 2 that can happen when local
central initiated PHY Update reserves 2 node rx, one for PHY
Update complete and another for Data Length Update complete
notification. Otherwise, a peripheral only needs 1
additional node rx to generate Data Length Update complete
when PHY Update completes.

Relates to #36381.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

avisconti committed:
drivers/sensor: lis2dw12: Preserve APIs error code

Return the same error code returned by HAL i/f APIs.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

cvinayak committed:
Bluetooth: Controller: Fix free rx buffer MFIFO enqueue

Replace the post decrement in loop's conditional into
explicit decrement inside the loop so as to avoid
decrementing the maximum count without enqueueing free rx
buffers into the free rx buffer MFIFO.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Commit:
net: mqtt: Add custom transport type

Add new custom transport type.
This allows user defined transport for MQTT communication.
The user must implement the transport procedure.

Fixes **#27015**

Signed-off-by: Vlad Tuhut <vlad.tuhut@raptor-technologies.ro>

mmahadevan108 committed:
dts: bindings: imx-gpio: Do not require interrupts

Some SoC's do not have interrupts connected to the GPIO module

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
drivers: igpio: Do not require GPIO interrupts to be present

Some SoC's do not have a GPIO interrupt connected

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

affrinpinhero-2356 committed:
board/STm32_Mini_dev_x_defconfig: ARM_MPU enabled without MPU.

This commit resolves warning ARM_MPU & HW_STACK_PROTECTION
during build. The MCU in STm32_Mini_dev (black & blue) has
no MPU support, but MPU was still enabled in defconfigs which
was the reason behind warnings. This commit removes
ARM_MPU & HW_STACK_PROTECTION from defconfig for both
STm32_Mini_dev development boards.

fix: zephyrproject-rtos/zephyr#36408

Signed-off-by: Affrin Pinhero <affrin.pinhero@hcl.com>

mnkp committed:
net: refactor tftp library

- bugfix: Accept initial tftp server reply from a port different than
  the one used to establish the connection (typically 69) as mandated
  by RFC 1350. Previous implementation was not standard compliant.
- bugfix: close socket in case of error or timeout.
- bugfix: Reset retransmit counter after receipt of a good packet.
- bugfix: Use CONFIG_TFTP_LOG_LEVEL to set log level.
- api: upon successful receipt of the file set `client.user_buf_size`
  to the size of the file received.
- Restructure the code, comments.
- Limit usage of global variables.
- Limit usage of `goto`.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

Damian-Nordic committed:
boards: thingy53: start network core before 802.15.4 driver

The nRF5 802.15.4 driver tries to establish a rpmsg channel
with the network core, so the network core must be started
earlier. Therefore, decrease the priority level of the task
responsible for initializing the board.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

pdgendt committed:
net: openthread: Convert API worker thread to a work queue

Direct openthread API usage requires explicit locking,
which is also used internally.
Exposing a work queue through the openthread context allows
work to be submitted without the need to block other threads.

In particular with CONFIG_OPENTHREAD_MANUAL_START, application
logic can offload work which otherwise would need to wait for
the lock to become available.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

gmarull committed:
doc: extensions: add kconfig directive and role support

Add a new extension to support the :kconfig: role and .. kconfig::
directive. This removes the need of using :option:.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
doc: use kconfig role and directive

Stop using :option: for Kconfig options.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

pdgendt committed:
net: coap: Fix coap client observe out-of-order messages

coap_response_received returned NULL if the observe option was out of
order, however it makes more sense to return the coap_reply handler
without actually calling it.

Additionally the reorder check has been modified to partially match
the RFC.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

gmarull committed:
doc: extensions: remove eager_only

The eager_only is currently breaking the usage of the `.. only``
directive, so remove it. Documentation seems to build fine without it,
so it has been removed for now.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
doc: doxygen: replace option alias with kconfig

Similar to Sphinx, @kconfig{} alias should be used in Doxygen docstring
in order to reference a Kconfig option. @option{} is still kept for
compatibility reasons.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

tarotene committed:
correct spells

tarotene committed:
correct spells

galak commented:
@tarotene thanks for the fix.  Can you please fix the commit message as per the projects Contribution Guidelines.  (Need a signed-off-by and commit message needs a header).

gmarull committed:
doc: release: add kconfig Sphinx and Doxygen changes

Add notes on Sphinx and Doxygen changes related to Kconfig referencing.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

mbolivar-nordic committed:
drivers: sensor: clean up zephyr_library calls

In drivers/sensor/CMakeLists.txt, we have various lines like this:

    add_subdirectory_ifdef(CONFIG_FOO foo)

Then drivers/sensor/foo/CMakeLists.txt says:

    zephyr_library()
    zephyr_library_sources_ifdef(CONFIG_FOO foo.c)

This is redundant; the foo/CMakeLists.txt won't be added to the build
system unless CONFIG_FOO=y in the first place, so there's no need for
extra boilerplate testing it again.

Remove all these unnecessary instances in each sensor driver's
CMakeLists.txt using this pattern:

    zephyr_library()
    zephyr_library_sources(foo.c)

In a couple of places, the '.c' extension is missing. Add them in for
consistency when that happens.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

benediktibk committed:
manifest: order subsystems alphabetically

Order the subsystems in the manifest file alphabetically, as they
actually should be according to the comment within the manifest file.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

galak committed:
tests: kernel: device: Exclude beaglev_starlight_jh7100

The beaglev_starlight_jh7100 uses a full 64-bit devicetree map
which uses #{address/size}-cells = 2.  The device test expects
that #{address/size}-cells = 1 so exclude beaglev_starlight_jh7100
from the test.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

tarotene committed:
Merge branch 'main' of https://github.com/tarotene/zephyr

tarotene commented:
Thank you for reviewing.

I've noticed that I haven't read the [guideline](https://docs.zephyrproject.org/latest/contribute/index.html) enough and haven't given a topic-branch-like name to my local branch, so I will close this PR and make another PR.
